addons/hr_timesheet/models/project.py

```diff
@@ -147,6 +147,20 @@ def write(self, values):
                 raise UserError(_('This task must be part of a project because there are some timesheets linked to it.'))
         return super(Task, self).write(values)
 
+    @api.multi
+    def name_get(self):
+        names = super().name_get()
+        if self.env.context.get('hr_timesheet_display_remaining_hours'):
+            name_mapping = dict(names)
+            for task in self:
+                if task.allow_timesheets and task.planned_hours > 0:
+                    mins = int((task.remaining_hours * 60) % 60)
+                    hours = int((task.remaining_hours * 60 - mins) // 60)
+                    hours_left = _(" %s:%s remaining") % (hours, mins)
+                    name_mapping[task.id] = name_mapping.get(task.id, '') + hours_left
+            names = list(name_mapping.items())
```

Comment:
direct return inside if 
```suggestion
            return list(name_mapping.items())
```